src/Unleash/DefaultUnleash.cs

```diff
@@ -118,7 +119,7 @@ public Variant GetVariant(string toggleName, UnleashContext context, Variant def
                 EmitImpressionEvent("getVariant", enhancedContext, variant.IsEnabled, toggleName, variant.Name);
             }
 
-            return Variant.UpgradeVariant(variant);
+            return variant;
```

Comment:
I don't think I'm sold on this one. Leaking a child dependency through a public API seems... well... very leaky. I don't think the cost of keeping just the Variant class around is particularly high. This PR is already very ambitious in the scope of things it removes... can we just keep it for now?

Reply:
Yep, after we've discussed this, I brought it back in https://github.com/Unleash/unleash-client-dotnet/pull/232/commits/c11ce8114b79642fc522f6ee6f4c2bedb151a877